app/scripts/modules/ecs/src/serverGroup/configure/wizard/capacityProvider/CapacityProvider.tsx

```diff
@@ -61,6 +89,24 @@ class EcsCapacityProvider extends React.Component<IEcsCapacityProviderProps, IEc
     this.setState({ capacityProviderStrategy: capacityProviderStrategy });
   };
 
+  private updateCapacityProviderStrategy = (targetCapacityProviderType: string) => {
+    const useDefaultCapacityProviders = targetCapacityProviderType == 'defaultCapacityProvider';
+    this.setState({useDefaultCapacityProviders : useDefaultCapacityProviders});
+    this.props.notifyAngular("useDefaultCapacityProviders", useDefaultCapacityProviders);
+
+    if (useDefaultCapacityProviders) {
+      this.setState({capacityProviderStrategy : []});
+      this.props.notifyAngular('capacityProviderStrategy', []);
+      if (this.state.defaultCapacityProviderStrategy.length > 0){
+        this.setState({capacityProviderStrategy : this.state.defaultCapacityProviderStrategy});
+        this.props.notifyAngular('capacityProviderStrategy', this.state.defaultCapacityProviderStrategy );
+      }
+    } else {
+      this.setState({capacityProviderStrategy : []});
+      this.props.notifyAngular('capacityProviderStrategy', []);
+    }
```

Comment:
There's duplicate logic in the if and else, and it's a bit confusing to follow. what do you think about doing something like this:

```
if (useDefaultCapacityProviders && this.state.defaultCapacityProviderStrategy.length > 0) {
    this.setState({capacityProviderStrategy : this.state.defaultCapacityProviderStrategy});
    this.props.notifyAngular('capacityProviderStrategy', this.state.defaultCapacityProviderStrategy );
} else {
    this.setState({capacityProviderStrategy : []});
    this.props.notifyAngular('capacityProviderStrategy', []);
}
```

Reply:
This logic make sense to me. After adding this logic the method becomes very compact and doesn't do much than if and else conditions. 

> It feels like this method is doing a lot. Can we separate this into an update useDefaultCapacityProviders method and a updateCapacityProviderStrategy method (you can involve updateCapacityProviderStrategy from within useDefaultCapacityProviders, but it feels like this method has too much responsibility


So I think keeping this logic in the same method makes better sense. What do you think ?